lib/angular2/shared/sockets/connections.ts

```diff
@@ -2,85 +2,156 @@
+        secure: false,
+        forceNew: true,
+        forceWebsockets: true
       });
-      this.connections[url].on('connect', () => {
-        if (!this.configured)
-          this.setupConnection(url, token, config);
+      // Listen for connection
+      this.on('connect', () => {
+        this.subjects.onConnect.next('connected');
+        // Authenticate or start heartbeat now    
+        this.emit('authentication', token);
       });
-      let forceConfig: any = setInterval(() => {
-        if (!this.configured && this.connections[url] && this.connections[url].connected) {
-          console.info('Forcing IO Configuration');
-          this.setupConnection(url, token, config);
-          clearInterval(forceConfig);
-        } else if (this.configured) {
-          clearInterval(forceConfig);
-        }
-      }, 1000)
-    } else {
-      console.log('Reusing existing connection: ', url);
+      // Listen for authentication
+      this.on('authenticated', () => {
+        this.subjects.onAuthenticated.next();
+        this.heartbeater();
+      })
+      // Listen for authentication
+      this.on('unauthorized', (err: any) => {
+        this.subjects.onUnAuthorized.next(err);
+      })
+      // Listen for disconnections
+      this.on('disconnect', (status: any) => this.subjects.onDisconnect.next(status));
+    } else if (this.socket && !this.socket.connected){
+      this.socket.off();
+      this.socket.destroy();
+      delete this.socket;
+      this.connect(token);
     }
-    return this.connections[url];
   }
-
-  public disconnect() {
-    Object.keys(this.connections).forEach((connKey) => {
-      if (this.connections[connKey] && this.connections[connKey].connected) {
-        this.connections[connKey].disconnect();
-      }
-      if (this.connections[connKey].off) {
-        this.connections[connKey].off();
-      }
-    });
-    this.connections = {};
-    this.configured  = false;
+  /**
+   * @method isConnected
+   * @description
+   * This method will return true or false depending on established connections
+   **/
+  public isConnected(): boolean {
+    return (this.socket && this.socket.connected);
   }
-
-  private setupConnection(url: string, token: AccessToken, config: any): void {
-    this.configured = true;
-    console.log('Connected to %s', url);
-    if (token.id) {
-      console.log('Emitting authentication', token.id);
-      this.connections[url].emit('authentication', token);
-    }
-    this.connections[url].on('unauthorized', (res: any) => console.error('Unauthenticated', res));
+  /**
+   * @method on
+   * @description
+   * This method will wrap the original "on" method and run it within the Angular Zone
+   **/
+  public on(event: string, handler: Function): void {
+    this.socket.on(event, (data: any) => this.zone.run(() => handler(data)));
+  }
+  /**
+   * @method emit
+   * @description
+   * This method will wrap the original "on" method and run it within the Angular Zone
```

Comment:
Cool thanks Ill fix it once I get a charger for my macbook, mine died yesterday :(